p2p/src/main/java/bisq/network/p2p/P2PService.java

```diff
@@ -336,12 +337,11 @@ public void onPreliminaryDataReceived() {
 
     @Override
     public void onUpdatedDataReceived() {
-        if (!isBootstrapped) {
-            isBootstrapped = true;
-            maybeProcessAllMailboxEntries();
-            p2pServiceListeners.stream().forEach(P2PServiceListener::onUpdatedDataReceived);
+        isBootstrapped = true;
+        maybeProcessAllMailboxEntries();
+        p2pServiceListeners.stream().forEach(P2PServiceListener::onUpdatedDataReceived);
+        if(!isBootstrapped)
             p2PDataStorage.onBootstrapComplete();
```

Comment:
The new version would behave different, even it might be not drastic and visible changes, I think that this is dangerous.

To make it more clear I post the old and the new method:
New version:
```
    public void onUpdatedDataReceived() {
        if (!isBootstrapped() || Capabilities.app.containsAll(Capability.SEED_NODE)) {
            maybeProcessAllMailboxEntries();
            p2pServiceListeners.stream().forEach(P2PServiceListener::onUpdatedDataReceived);
        }

        if (!isBootstrapped()) {
            isBootstrapped = true;
            p2PDataStorage.onBootstrapComplete();
        }
    }
```
Old version:
```
    public void onUpdatedDataReceived1() {
        if (!isBootstrapped) {
            isBootstrapped = true;
            maybeProcessAllMailboxEntries();
            p2pServiceListeners.stream().forEach(P2PServiceListener::onUpdatedDataReceived);
            p2PDataStorage.onBootstrapComplete();
        }
    }
```

  isBootstrapped = true; is set after maybeProcessAllMailboxEntries(); is called, but there we check for isBootstrapped so that will behave differently. the flag need to be set initially before any other calls.


`p2pServiceListeners.stream().forEach(P2PServiceListener::onUpdatedDataReceived);`

I am not sure if a repated onUpdatedDataReceived notifiaction might cause issues. It is expected to be only called once. I know its for seed node only and likely will not have effects, but I think it would be better to add a new handler (can be default in the interface so it will not require clients to implement new handler method).

And a minor issue:
`isBootstrapped()` should be replaced by `isBootstrapped`.

Reply:
I am running a long-time test with the original code. It seems that everything works just fine without the change. God I love when minor changes somewhere in core change the behavior elsewhere without any trace :).